Add role ocp4_workload_synopsys_blackduck

davemeurer opened the pull request:
##### SUMMARY

<!--- Describe the change below, including rationale and design decisions. -->
(Opening as a DRAFT PR) This role deploys the Synopsys Black Duck integration and a demo application on an OpenShift Workshop. This is part of an effort to add RHPDS Catalog entries for our Partners, and provide the ability to perform demos and workshops. 

##### ISSUE TYPE
- New role Pull Request

##### COMPONENT NAME
ocp4_workload_synopsys_blackduck

##### ADDITIONAL INFORMATION
The workshop guide for users will be published on demo.openshift.com
I ran yamllint and here are the results:
```defaults/main.yml
files/insecure-bank-deployment.yml
files/insecure-bank-route.yml
files/insecure-bank-service.yml
tasks/main.yml
tasks/post_workload.yml
tasks/pre_workload.yml
tasks/remove_workload.yml
tasks/workload.yml
tasks/workload.yml
  13:17     warning  truthy value should be one of [false, true]  (truthy)
```
<!--- Paste verbatim command output below, e.g. before and after your change -->
<!-- ansible --version -->
<!-- pip freeze -->
```paste below
ansible 2.9.11
  config file = None
  configured module search path = ['/Users/dmeurer/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.9.11/libexec/lib/python3.8/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.8.5 (default, Jul 21 2020, 10:48:26) [Clang 11.0.3 (clang-1103.0.32.62)]
```

wkulhanek commented on ansible/roles_ocp_workloads/ocp4_workload_synopsys_blackduck/tasks/workload.yml:
```diff
@@ -0,0 +1,82 @@
+    - ./files/insecure-bank-service.yml
+    - ./files/insecure-bank-route.yml
+
+- set_fact:
```
1. Name!
2. I see...

davemeurer replied:
Added name

davemeurer commented on ansible/roles_ocp_workloads/ocp4_workload_synopsys_blackduck/tasks/workload.yml:
```diff
@@ -0,0 +1,82 @@
+    creates: /usr/bin/synopsysctl
+
+- set_fact:
+    ocp4_workload_current_project: "{{ ocp4_workload_synopsys_blackduck_demo_project }}"
```
1. Ah, thanks @wkulhanek !! I'll add names asap.
2. Since I'm creating 2 namespaces, I wanted to use the same project.j2 template. Happy to entertain a better way?

davemeurer replied:
Added name

wkulhanek commented:
@davemeurer Is this still draft? Or should it be merged?

jkupferer commented on ansible/roles_ocp_workloads/ocp4_workload_synopsys_blackduck/tasks/workload.yml:
```diff
@@ -0,0 +1,84 @@
+- name: create {{ ocp4_workload_synopsys_blackduck_project }} project for user by using a jinja2 template in the role
+  k8s:
+    state: present
+    definition: "{{ lookup('template', role_path ~ '/templates/project.j2' ) | from_yaml }}"
```
Template lookups already default to looking in the template directory under the role path, so this can be simplified to:
```
    definition: "{{ lookup('template', 'project.j2' ) | from_yaml }}"
```

davemeurer commented:
Thanks for the approval @wkulhanek - how would I go about getting this merged?

wkulhanek commented:
@davemeurer You said this was a draft PR. Is it still draft? If not I can merge it.

davemeurer commented:
Hi @wkulhanek, I initially placed it in draft, but after addressing your suggestions, I clicked the button to turn off draft status. Please do merge. Thanks!
